packages/scan-request-sender/src/sender/dispatcher.ts

```diff
@@ -13,10 +14,11 @@ export class Dispatcher {
         @inject(PageDocumentProvider) private readonly pageDocumentProvider: PageDocumentProvider,
         @inject(Logger) private readonly logger: Logger,
         @inject(ScanRequestSender) private readonly sender: ScanRequestSender,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
     ) {}
 
     public async dispatchScanRequests(): Promise<void> {
-        const configQueueSize = Number(process.env.QUEUE_SIZE);
+        const configQueueSize = (await this.serviceConfig.getConfigValue('commonConfig')).maxQueueSize;
```

Comment:
>'commonConfig' [](start = 73, length = 14)

Can we use generic instead of string?
serviceConfig.getConfigValue<configType>().value

Reply:
even though we are using string, it is type safe. So, if you call serviceConfig.getConfigValue('blah'), it will fail

Reply:
also, i didnt want to add a method for every property in the service config.

Reply:
how we discover the name then?

---
In reply to: [304637452](https://github.com/microsoft/accessibility-insights-service/pull/210#discussion_r304637452) [](ancestors = 304637452)